### DISCUSSION TITLE
Using Wemos D1 Mini

### DISCUSSION THREAD

**Discussion by nikandriko** [phord/Jarvis#15] [Show and tell]
Did a new Pebble board using the D1 Mini board based on the [RJ-45 pin-label mapping](https://github.com/phord/Jarvis#physical-interface) and the [D1 pinout mapping](https://github.com/phord/Jarvis/commit/e859bc3233684c0186226da7aab489438b025065). 

![image](https://user-images.githubusercontent.com/962911/152632371-ee4c47ee-6f86-45ee-9765-d47264495863.png)

This is the configuration I used on Pebble (in case anyone wants to load it on [Pebble](http://pebble.my-board.org/) and modify it).

```
Terminal||38|411|1||TS?||8|4|Terminals||terminal_481
Terminal||38|438|1||TS?||8|4|Terminals||terminal_481
Note||30|524|2||||2||RJ-45 Connectors||NOTEPAD_22
Wire||101|312|21||11|#000000|3|11||10|
Wire||155|339|21||11|#FF0000|2|11||10|
Wire||371|339|21||11|#FF0000|2|11||10|
Terminal||281|411|1||TS?||6|4|Terminals||terminal_461
Wire||425|312|21||11|#000000|3|11||10|
Note||281|525|2||||2||FTDI Connection||NOTEPAD_22
Terminal||577|438|1||TS?||8|4|Terminals||terminal_481
Terminal||577|658|1||TS?||8|4|Terminals||terminal_481
Note||606|522|2||||2||Wemos D1 Mini||NOTEPAD_22
Wire||586|339|21||11|#FF0000|2|11||10|
Wire||613|312|21||11|#000000|3|11||10|
Wire||76|475|13||11|#3253FF|10|11||10|
Wire||130|502|13||11|#A233C3|7|11||10|
Wire||319|466|12||11|#A233C3|13|11||10|
Wire||346|505|13||11|#3253FF|13|11||10|
Wire||49|493|12||11|#FFFF00|25|11||10|
Wire||182|504|21||11|#FFFF00|3|11||10|
Wire||209|504|21||11|#FFFF00|3|11||10|
Wire||236|504|21||11|#FFFF00|3|11||10|
Wire||184|695|11||11|#FFFF00|18|11||10|
Wire||211|640|11||11|#FFFF00|19|11||10|
Wire||238|612|11||11|#FFFF00|17|11||10|
BREADBOARDSTYLE=BB12
SHOWTHETOPAREA=true
```
Please let me know if I'm missing anything.

**Comment by phord** [comment]
- The Wemos board is wider than the Oak board, so you need to move the upper header from row H to row I.
- The Wemos D1 does not need the FTDI interface because it has an RS232 interface in the USB connector.
- The LED is on the other side of the board from the one you showed.  On my Wemos buiId, I flipped it over so the LED is visible for debugging and so the USB connector faces off the end of the board for easier connection.

This is how my Wemos board is wired:
```Terminal||38|411|1||TS?||8|4|Terminals||terminal_481
Terminal||38|438|1||TS?||8|4|Terminals||terminal_481
Note||30|524|2||||2||RJ-45 Connectors||NOTEPAD_22
Wire||101|312|21||11|#000000|3|11||10|
Wire||155|339|21||11|#FF0000|2|11||10|
Terminal||631|411|1||TS?||8|4|Terminals||terminal_481
Terminal||631|658|1||TS?||8|4|Terminals||terminal_481
Note||706|516|2||||2||wifi-[Wemos D1 Mini]-usb||NOTEPAD_22
Wire||829|339|21||11|#FF0000|2|11||10|
Wire||802|312|21||11|#000000|3|11||10|
Wire||76|478|13||11|#3253FF|24|11||10|
Wire||130|466|12||11|#A233C3|23|11||10|
Wire||49|383|12||11|#FFFF00|24|11||10|
Wire||182|504|21||11|#FFFF00|3|11||10|
Wire||209|504|21||11|#FFFF00|3|11||10|
Wire||236|504|21||11|#FFFF00|3|11||10|
Wire||184|695|11||11|#FFFF00|21|11||10|
Wire||211|640|11||11|#FFFF00|18|11||10|
Wire||238|612|11||11|#FFFF00|18|11||10|
BREADBOARDSTYLE=BB12
SHOWTHETOPAREA=true
```

![image](https://user-images.githubusercontent.com/123908/152662008-041759e1-86ef-416a-a469-8f8aa463c14a.png)

I actually modified my original board (which was wired for the Digistump Oak) to take the Wemos D1 Mini. So in my board, you can see the FTDI header is still there, and I added a 2nd row of headers to accommodate the wider Wemos.
![PXL_20220205_225133877](https://user-images.githubusercontent.com/123908/152662194-b77dcfc8-4c7c-4519-ad53-554e70b769a2.jpg)

Here it is with the Wemos board attached.
![PXL_20220205_225101558](https://user-images.githubusercontent.com/123908/152662198-e53d51d0-f374-4493-a209-90e2412398a0.jpg)

**Reply by nikandriko** [reply]
Thanks Phil (@phord), super helpful!
Would you mind also sharing the IFTTT routines you've set? 
What do you publish on the MQTT channel?

**Comment by phord** [comment]
These are my IFTTT applets:
[If You say "Stand up", then Send data to Jarvis - preset feed](https://ifttt.com/applets/r2zeGKfq-if-you-say-stand-up-then-send-data-to-jarvis-preset-feed)
[If You say "Sit down", then Send data to Jarvis - preset feed](https://ifttt.com/applets/A3WD8MtA-if-you-say-sit-down-then-send-data-to-jarvis-preset-feed)

As mentioned in the README.md, I send data to two feeds at Adafruit.  When I press a preset button (1-4) I send that number to the "preset" feed.  When the desk changes height, I send the height number (in mm or inches) to the "height" feed.  You can see the feeds by following the links in the README.md.

**Reply by nikandriko** [reply]
@phord the IFTTT applets do not open in the links. Would you mind sharing some screenshots? Basically, I'm trying to understand what the applets publish in the MQTT feeds. I suspect they each publish a number to the "preset" feed.

**Reply by phord** [reply]
It would be nice if these were exportable.
![image](https://user-images.githubusercontent.com/123908/153114261-7e2588db-c583-41e3-af32-1b04ac17ce9b.png)
![image](https://user-images.githubusercontent.com/123908/153114286-0e1eb5d9-6225-4f4e-9e12-797d73ea2333.png)
![image](https://user-images.githubusercontent.com/123908/153114294-6668c0ce-835c-4848-8afa-671c59a2a954.png)
